### PR TITLE
CR-3573: Added terraform working directory optional setting

### PR DESCRIPTION
**Pull request by Tomerva**

### CODE REVIEW

File: src/commands/scan.ts [hunk 1]
```diff
@@ -131,7 +133,19 @@ async function getVcsInfo(baseDir: string): Promise<VcsInfo | undefined> {
      return vcsInfo;
 }
 
-async function getTerraformWorkingDirectory(): Promise<string | undefined> {
+async function getTerraformWorkingDirectory(config: CloudrailConfiguration) {
+    const directoryFromConfig = getTerraformWorkingDirectoryFromConfig(config);
```
Comment (Viperain) [inline src/commands/scan.ts:137]: @Tomerva  is weird to me that getTerraformWorkingDirectoryFromConfig use config but that you have the validation "else if (config.terraformWorkingDirectory)" which also access config.. its not encapsulated who have access to terraformWorkingDirectory. i think cleaner way is 
```
async assessmentStart(): Promise<void> {
        this.resetView('Scanning, please wait...');
        const workingDir = config.terraformWorkingDirectory;
        if (workingDir) {
            const directoryFromConfig = getTerraformWorkingDirectoryFromConfig(workingDir);
            if (directoryFromConfig) {
                return directoryFromConfig;
            } else {
                vscode.window.showErrorMessage('The directory specified in the TerraformWorkingDirectory settings could not be resolved. Either specify an absolute path or leave empty to scan on the active editor\'s directory.');
                return;
            }
        } else {
            return await getTerraformWorkingDirectoryFromActiveEditor();
        }
    }
```

and just change getTerraformWorkingDirectoryFromConfig get string. there is no real value in sending the whole config

File: src/run_result_handlers/sidebar/cloudrail_tree_item.ts [hunk 2]
```diff
@@ -54,4 +54,10 @@ export class CloudrailIssueItemTreeItem extends CloudrailTreeItem {
             ]
         };
     }
+}
+
+export class NotificationTreeItem extends CloudrailTreeItem {
```
Comment (Viperain) [inline src/run_result_handlers/sidebar/cloudrail_tree_item.ts:59]: @Tomerva  how its related to the pr?

**Comment by Viperain** [comment]
@Tomerva  two small comments. also - can u add the pr screenshots of any UI related changes? thanks

**Review by Viperain — CHANGES_REQUESTED** [review]

File: src/run_result_handlers/sidebar/cloudrail_tree_item.ts (hunk 2 above)
Comment (Tomerva) [reply src/run_result_handlers/sidebar/cloudrail_tree_item.ts:59]: its not really.. didnt want to open a separate ticket for something so small

**Review by Viperain — APPROVED** [review]

**Pull request merged**